conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/DialogueInterfaceGenerator.java

```diff
@@ -166,7 +166,15 @@ private MethodSpec apiMethod(EndpointDefinition endpointDef, Function<Optional<T
                 .forEach(referenceType -> methodBuilder.addAnnotation(
                         ClassName.get(referenceType.getPackage(), referenceType.getName())));
 
-        endpointDef.getDeprecated().ifPresent(deprecatedDocsValue -> methodBuilder.addAnnotation(Deprecated.class));
+        endpointDef.getDeprecated().ifPresent(_deprecatedValue -> {
+            if (endpointDef.getTags().contains("deprecated-for-removal")) {
+                methodBuilder.addAnnotation(AnnotationSpec.builder(Deprecated.class)
+                        .addMember("forRemoval", "true")
```

Comment:
We could put this line into a conditional on `endpointDef.getTags().contains("deprecated-for-removal")` and reuse the `methodBuilder.addAnnotation(AnnotationSpec.builder(Deprecated.class).build()` structure in both cases, but that requires holding a reference to the AnnotationSpec.Builder, which isn't particularly fluent.
lgtm.